docs/tutorials/onnx/inference_on_onnx_model.md

```diff
@@ -0,0 +1,259 @@
+images = ['apron', 'hammerheadshark', 'dog', 'wrench', 'dolphin', 'lotus']
+base_url = "https://github.com/ThomasDelteil/web-data/blob/c77c2e93ba142f45682ed63c191d2568b20aff25/mxnet/doc/tutorials/onnx/{}?raw=true"
+
+if not os.path.isdir(image_folder):
+    os.makedirs(image_folder)
+    for image in images: wget.download(base_url.format("{}/{}.jpg".format(image_folder, image)), image_folder)
+if not os.path.isfile(utils_file):
+    wget.download(base_url.format(utils_file))                               
+```
+
+
+```python
+from utils import *
+```
+
+## Downloading a model from the [ONNX model zoo](https://github.com/onnx/models)
+
+We download a pre-trained model, in our case the [vgg16](https://arxiv.org/abs/1409.1556) model, trained on [ImageNet](www.image-net.org/). The model comes packaged in an archive `tar.gz` file containing an `model.onnx` model file and some sample input/output data.
+
+
+```python
+base_url = "https://s3.amazonaws.com/download.onnx/models/" 
+current_model = "vgg16"
+model_folder = "model"
+archive = "{}.tar.gz".format(current_model)
+archive_file = os.path.join(model_folder, archive)
+url = "{}{}".format(base_url, archive)
+```
+
+Create the model folder and download the zipped model
+
+
+```python
+os.makedirs(model_folder, exist_ok=True)
+if not os.path.isfile(archive_file):  
+    wget.download(url, model_folder)
+```
+
+Extract the model
+
+
+```python
+if not os.path.isdir(os.path.join(model_folder, current_model)):
+    tar = tarfile.open(archive_file, "r:gz")
+    tar.extractall(model_folder)
+    tar.close()
+```
+
+The models have been pre-trained on ImageNet, let's load the label mapping of the 1000 classes.
+
+
+```python
+categories = json.load(open('ImageNet_labels.json', 'r'))
+```
+
+## Loading the model into MXNet Gluon
+
+
+```python
+onnx_path = os.path.join(model_folder, current_model, "model.onnx")
+```
+
+We get the symbol and parameter objects
+
+
+```python
+sym, params = onnx_mxnet.import_model(onnx_path)
+```
+
+We pick a context, CPU or GPU
+
+
+```python
+ctx = mx.cpu()
+```
+
+And load them into a MXNet Gluon symbol block
+
+
+```python
+net = gluon.nn.SymbolBlock(outputs=sym, inputs=mx.sym.var('input_0'))
+net_params = net.collect_params()
+for param in params:
+    if param in net_params:
+        net_params[param]._load_init(params[param], ctx=ctx)
+```
+
+This [hybridizes](https://mxnet.incubator.apache.org/tutorials/gluon/hybrid.html) the network the computational graph gets cached and we get performance gains
+
+
+
+```python
+net.hybridize()
+```
+
+## Test using sample inputs and outputs
+The model comes with sample input/output we can use to test that whether model is correctly loaded
+
+
+```python
+numpy_path = os.path.join(model_folder, current_model, 'test_data_0.npz')
+sample = np.load(numpy_path, encoding='bytes')
+inputs = sample['inputs']
+outputs = sample['outputs']
+```
+
+
+```python
+print("Input format: {}".format(inputs[0].shape))
```

Comment:
What's the output format? Why do you use `outputs[0]` when you're using argmax during testing?

Reply:
not sure what you mean? Do you mean I should show that the ouput is (1000,1) ?

Reply:
Yes, showing the shape helps. But more specifically, I'm referring to this line:
`print("Loaded model and sample output predict the same class: {}".format(np.argmax(result) == np.argmax(outputs[0])))`
Here you're using outputs[0] for all the results. What's the type of `outputs`? What are other indices?